astropy/stats/funcs.py

```diff
@@ -1007,6 +1009,152 @@ def bootstrap(data, bootnum=100, samples=None, bootfunc=None):
+    >>> resamples = jackknife(data)
+    >>> resamples
+    array([[ 2.,  3.,  4.,  5.,  6.,  7.,  8.,  9.,  0.],
+           [ 1.,  3.,  4.,  5.,  6.,  7.,  8.,  9.,  0.],
+           [ 1.,  2.,  4.,  5.,  6.,  7.,  8.,  9.,  0.],
+           [ 1.,  2.,  3.,  5.,  6.,  7.,  8.,  9.,  0.],
+           [ 1.,  2.,  3.,  4.,  6.,  7.,  8.,  9.,  0.],
+           [ 1.,  2.,  3.,  4.,  5.,  7.,  8.,  9.,  0.],
+           [ 1.,  2.,  3.,  4.,  5.,  6.,  8.,  9.,  0.],
+
+           [ 1.,  2.,  3.,  4.,  5.,  6.,  7.,  8.,  9.]])
+    >>> resamples.shape
+    (10, 9)
+
+    2. Obtain Jackknife estimate for the mean, its bias, its standard error, 
+    and its 95% confidence interval:
+    
+    >>> test_statistic = np.mean
+    >>> estimate, bias, stderr, conf_interval = jackknife(data, test_statistic, 0.95)[1:5]
+    >>> estimate
+    4.5
+    >>> bias
+    0.0
+    >>> stderr
+    0.95742710775633832
+    >>> conf_interval
+    array([ 2.62347735,  6.37652265])
+
+    3. Example for two estimates
+    
+    >>> test_statistic = lambda x: (np.mean(x), np.var(x))
+    >>> estimate, bias, stderr, conf_interval = jackknife(data, test_statistic, 0.95)[1:5]
+    >>> estimate
+    array([ 4.5       ,  9.16666667])
+    >>> bias
+    array([ 0.        , -0.91666667])
+    >>> stderr
+    array([ 0.95742711,  2.69124476])
+    >>> conf_interval
+    array([[  2.62347735,   3.89192387],
+           [  6.37652265,  14.44140947]])
+
+    IMPORTANT: Note that confidence intervals are given as columns
+
+    References
+    ---------
+    .. [1] McIntosh, Avery. "The Jackknife Estimation Method".
+        <http://people.bu.edu/aimcinto/jackknife.pdf>   
+
+    .. [2] Efron, Bradley. "The Jackknife, the Bootstrap, and other 
+        Resampling Plans". Technical Report No. 63, Division of Biostatistics, 
+        Stanford University, December, 1980.
+
+    .. [3] Cowles, Kate. "Computing in Statistics: The Jackknife, Lecture 11". 
+        <http://homepage.stat.uiowa.edu/~kcowles/s166_2009/lect11.pdf>. 
+        September, 2009.
+    """
+
+    n = data.shape[0] 
+    assert n > 0, "data must contain at least one measurement" 
+    resamples = np.empty([n,n-1])
+
+    for i in range(n):
+        resamples[i] = np.delete(data, i)
+
+    if statistic is not None:
+        stat_data = statistic(data)
+        jack_stat = np.apply_along_axis(statistic,1,resamples)
+        mean_jack_stat = np.mean(jack_stat,axis=0)
+        
+        # jackknife bias
+        bias = (n-1)*(mean_jack_stat - stat_data)
+        
+        # jackknife standard error
+        std_err = np.sqrt((n-1)*np.mean((jack_stat - mean_jack_stat)*(jack_stat - mean_jack_stat),axis=0))
+        
+        # bias-corrected "jackknifed estimate"
+        estimate = stat_data - bias
+
+        # jackknife confidence interval
+        if conf_lvl is not None:
+            assert (conf_lvl > 0 and conf_lvl < 1), "confidence level must be in (0,1)." 
+            from scipy.special import erfinv    
```

Comment:
This has been the style that has been used for other functions in this module but this should be fixed at some point to match the [coding guidleines](http://docs.astropy.org/en/stable/development/codeguide.html).   Since it is an easy fix, I'd suggest @mirca updates this function to match the requirements.